code/modules/modular_computers/file_system/programs/statusdisplay.dm

```diff
@@ -15,36 +15,76 @@
 	var/upper_text
 	var/lower_text
 
-/datum/computer_file/program/status/proc/SendSignal()
+/**
+ * Post status display radio packet.
+ * Arguments:
+ * * command - the status display command
+ * * data1 - the data1 value, as defined by status displays
+ * * data2 - the data2 value, as defined by status displays
+ */
+/datum/computer_file/program/status/proc/post_status(command, data1, data2)
 	var/datum/radio_frequency/frequency = SSradio.return_frequency(FREQ_STATUS_DISPLAYS)
-
 	if(!frequency)
 		return
 
-	var/datum/signal/status_signal = new(list("command" = "message"))
+	var/datum/signal/status_signal = new(list("command" = command))
+	switch(command)
+		if("message")
+			status_signal.data["msg1"] = data1
+			status_signal.data["msg2"] = data2
+		if("alert")
+			status_signal.data["picture_state"] = data1
```

Comment:
```suggestion
		if("message")
			status_signal.data.top_text = data1
			status_signal.data.bottom_text = data2
		if("alert")
			status_signal.data.picture_state = data1
```

Reply:
I realize that you weren't the one who initially wrote the code for this, but I'll go ahead and nitpick so that the code is brought up to standards.

It's better to use the `.` to refer to the key in a list directly.

Using `status_signal.data.top_text` is suggested instead of `status_signal.data["top_text"]` since it is _slightly_ less verbose and is predominantly the way lists are accessed throughout the repo.  

Situations where people use `[""]` to access the list is when there are spaces in the key.  As an example `status_signal.data["top text"]` would be when you use  `[""]` 
 to access a key since `status_signal.data.top text` would cause an error.  I hope that makes sense.

I'd also suggest you replace `["msg1"]` with `top_text` and `["msg2"]` with `bottom_text` so they are better named.  They also need to be changed in other files located at the following locations:

https://github.com/tgstation/tgstation/blob/415c26dea11ac0d2ed53f1ea420ebdcb245dc860/code/game/machinery/computer/communications.dm#L744-L750

https://github.com/tgstation/tgstation/blob/8c7e4ef6ee0b6e60783ccb084d60138353b2e2be/code/game/machinery/status_display.dm#L346-L348

Reply:
done